src/fpm/manifest/dependency.f90

```diff
@@ -219,20 +223,44 @@ subroutine new_dependencies(deps, table, root, error)
 
         type(toml_table), pointer :: node
         type(toml_key), allocatable :: list(:)
-        integer :: idep, stat
+        logical, allocatable :: non_meta(:)
+        integer :: idep, stat, ndep
 
         call table%get_keys(list)
         ! An empty table is okay
         if (size(list) < 1) return
 
-        allocate (deps(size(list)))
+        !> If requesting metapackages, do not stop on meta keywords
+        if (present(meta)) then
+            ndep = 0
+            do idep = 1, size(list)
+                if (is_meta_package(list(idep)%key)) cycle
+                ndep = ndep+1
+            end do
+
+            !> Return metapackages config from this node
+            call new_meta_config(meta, table, error)
+            if (allocated(error)) return
+
+        else
+            ndep = size(list)
+        end if
+
+        allocate(deps(ndep))
+        ndep = 0
         do idep = 1, size(list)
+
+            ! Skip meta packages
```

Comment:
Aren't you already skipping meta-packages [here](https://github.com/fortran-lang/fpm/pull/859/files#diff-2a797402d67a1a4dc0a46a1fee5ea061ab7966b0205f4d6c3f8eab4b1cbc60b8R237) ?